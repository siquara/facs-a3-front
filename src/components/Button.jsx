import React from "react";
export function Button({ active, ...props }){
    return(
        <button 
        data-testid='botao'
        type="submit" 
        className={`
        ${active ? "bg-primary text-white" : "bg-white text-primary"} 
        font-[500] flex items-center justify-center break-words min-w-[290px] 
        max-w-[290px] h-[80px] px-[45px] mx-[7px] 
        border-solid border border-[#1E3954] rounded-full 
        hover:text-white hover:bg-primary hover:border-secondary transition-all 
        duration-500 ease-in-out
        `}
        {...props}
        >
            {props.children}
        </button>
    )
}

