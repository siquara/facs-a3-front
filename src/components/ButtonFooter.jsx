import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowUp } from "@fortawesome/free-solid-svg-icons";

const ButtonFooter = () => {
  const [isHovered, setIsHovered] = useState(false);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth"
    });
  };

  return (
    <div className="relative">
      <button
        onClick={scrollToTop}
        className="absolute bottom-11 right-5 bg-gray-700 text-white p-2 rounded-md focus:outline-none xs:bottom-[110px] xs:right-[-160px] sm:bottom-[110px] sm:right-[-170px] md:bottom-11 md:right-16 lg:bottom-11 lg:right-20 "
      >
        <div
          className={`xs:w-8 xs:h-14 sm:w-16 sm:h-52 bg-white rounded-tr-full rounded-tl-full flex items-center justify-center shadow-md ${isHovered ? 'bg-neutral-300 shadow-black ' : ''}`}
          onMouseEnter={() => setIsHovered(true)}
          onMouseLeave={() => setIsHovered(false)}
        >
          <div className="text-stone-600 text-2xl xs:mb-13 sm:mb-32">
            <FontAwesomeIcon icon={faArrowUp} className={isHovered ? 'text-black text-4xl' : 'text-stone-700 '} />
          </div>
        </div>
      </button>
    </div>
  );
};

export default ButtonFooter;
