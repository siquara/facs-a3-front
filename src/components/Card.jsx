import React from 'react';
import { ArrowRight } from '@phosphor-icons/react';
import PropTypes from 'prop-types';
export function Card({ title, description, image, link }) {
    return (
        <div data-testid="card" className="flex flex-col bg-[#ededed] md:bg-[#f5f5f5] shadow-md rounded-3xl py-12 px-9 xs400:px-12 transition duration-300 ease-in-out transform hover:bg-white mb-10 max-w-lg h-86 hover:shadow-[0_4px_4px_0px_rgba(0,0,0,0.3)]">
            <div data-testid="card_conteudo" className='mb-4'>
                <div className="flex items-center gap-2">
                    <img data-testid="card_conteudo_imagem" src={image} className='min-h-5 max-w-[300px]' />
                </div>
                <h1 data-testid="card_conteudo_titulo" className="text-3xl font-bold mt-5 text-secondary">{title}</h1>
                <p data-testid="card_conteudo_descricao" className="text-secondary text-base mt-5 hyphens-auto text-justify " >
                    {description}
                </p>
            </div>
            <div data-testid="card_botao" className="flex justify-center items-center mt-auto">
                <a data-testid="card_botao_link" href={link} target="_blank" className="flex gap-4 bg-secondary hover:bg-white hover:cursor-pointer hover:text-secondary hover:border hover:border-secondary text-white font-bold py-2 px-5 xs400:px-10 rounded-tr-3xl rounded-bl-2xl transition-all duration-300 ease-in-out">
                    Acesse aqui
                    <ArrowRight size={24} />
                </a>
            </div>
        </div>
    );
}

Card.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
};