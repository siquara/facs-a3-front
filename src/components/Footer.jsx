import LogoAnima from "/LogoAnima.png";
import LogoUnifacs from "/LogoUnifacs.png";

export function Footer (){
    return (
        <div className="bg-secondary text-white h-[568px] flex flex-col justify-between">
            <div className="">INFOS</div>
            <footer className="border-t flex flex-row justify-between items-center py-[24px] px-[120px]">
                <div className="font-[600]">UNIVERSIDADE:</div>
                <div className="flex flex-row gap-20">
                    <a href="/">
                    <img src={LogoAnima} alt="LogoAnima" className="max-w-[80%] md:max-w-[100%] w-[109px]"/>
                    </a>
                    <a href="/">
                        <img src={LogoUnifacs} alt="LogoUnifacs" className="max-w-[80%] md:max-w-[100%] w-[160px]"/>
                    </a>
                </div>
                
            </footer>
        </div>

    );
}