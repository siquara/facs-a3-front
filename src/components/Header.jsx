// eslint-disable-next-line no-unused-vars
import React from 'react';
import hermesLogo from "/assets/hermesLogo.svg";

export function Header() {
  return (
    <header className="bg-white rounded-e-full w-fit py-[12px] pe-[5px] lg:pe-[45px] ps-[20px] lg:ps-[120px] z-10 absolute top-[20px] lg:top-[50px]">
      <a href="./index.html"><img src={hermesLogo} alt="Logotipo do Hermes" className="max-w-[80%] lg:max-w-[100%]"/></a>
    </header>
  );
}
