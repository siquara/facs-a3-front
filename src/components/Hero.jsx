import HeroWoman from "/HeroWoman.png";
export function Hero() {
  return (
    <div className="hero h-screen bg-[url('/HeroImage.png')] bg-no-repeat bg-cover flex items-center">
      <div className="grid mx-auto max-w-7xl px-6 lg:gap-8 xl:gap-0 lg:grid-cols-12 items-center justify-center xs400:pt-[8rem] md:pt-[15rem] lg:pt-0">
        <div className="lg:col-span-7 ">
          <h2 className="text-white text-xl font-[300]">
            Bem-vindo(a) a <strong className="font-[600]">Hermes</strong>
          </h2>
          <h1 className="text-white text-4xl lg:text-5xl font-[700]">
            Encontre as Ferramentas ideais para vocÃª
          </h1>
        </div>
        <div className="lg:col-span-5 place-self-center ">
          <img src={HeroWoman} alt="Woman" className="md:max-w-[85%] lg:max-w-[100%]" />
        </div>
      </div>
    </div>
  );
}
