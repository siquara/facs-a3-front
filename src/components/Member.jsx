import React from 'react';
import PropTypes from 'prop-types';
import { Circle, GithubLogo,LinkedinLogo } from "@phosphor-icons/react";

export function Member({name, githubLink, linkedinLink}){
    return (
    <p className="flex items-center gap-2 w-full justify-between">
        {name}
        <Circle size={5} className="bg-white rounded-full" />
        <a
          href={githubLink}
          className="border-b-[2px] border-solid border-transparent hover:border-white"
          target="_blank"
        >
          <GithubLogo size={30} />
        </a>
        <a
          href={linkedinLink}
          className="border-b-[2px] border-solid border-transparent hover:border-white"
          target="_blank"
        >
          <LinkedinLogo size={30} />
        </a>
      </p>
    )
}

Member.propTypes = {
    name: PropTypes.string.isRequired,
    githubLink: PropTypes.string.isRequired,
    linkedinLink: PropTypes.string.isRequired,
};