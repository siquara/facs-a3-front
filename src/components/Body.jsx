import * as React from "react";
import { Card } from "./Card";
import { Button } from "./Button";
import { filterData } from "../utils/data/filter";
import { useKeenSlider } from "keen-slider/react";
import "keen-slider/keen-slider.min.css";
import { cards } from "../utils/data/cards";

export function Body() {
  const [sliderRef] = useKeenSlider({
    loop: false,
    mode: "free-snap",
    slides: {
      perView: "auto",
      spacing: 12,
    },
  });

  const [activeFilter, setActiveFilter] = React.useState(filterData[0].title);

  const [activeCards, setActiveCards] = React.useState(cards[filterData[0].id]);

  function handleChangeActiveCard(filter) {
    setActiveFilter(filter.title)
    setActiveCards(cards[filter.id]);
  }

  return (
    <div className="bg-white relative">
      <div className=" z-10 transform -translate-y-[40px]">
        <div ref={sliderRef} className="keen-slider w-full overflow-hidden">
          {filterData.map((filter, index) => (
            <div key={index} className="keen-slider__slide min-w-fit max-w-fit">
              <Button
                active={activeFilter === filter.title}
                onClick={() => handleChangeActiveCard(filter)}
              >
                {filter.title}
              </Button>
            </div>
          ))}
        </div>
      </div>
      <div className=" flex flex-wrap gap-10 justify-center mx-auto max-w-7xl px-6">

        {
          activeCards.map((filter, index) => {
            return (
              <Card
                key={index}
                title={filter.title}
                description={filter.description}
                image={filter.image}
                link={filter.link}
              />
            );
          })
        }
      </div>
    </div>
  );
}
