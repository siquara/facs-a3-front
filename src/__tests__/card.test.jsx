import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Card } from '../components/Card';

describe('Card Component', () => {
    const props = {
        title: 'Titulo Teste',
        description: 'Descrição Teste',
        image: 'teste.png',
        link: 'http://localhost/'
    };

    it('Renderizar com os props definidos', () => {
        render(<Card {...props} />);

        expect(screen.getByText(props.title)).toBeInTheDocument();
        expect(screen.getByText(props.description)).toBeInTheDocument();
        expect(screen.getByRole('link')).toHaveAttribute('href', props.link);

        const imageElement = screen.getByRole('img');
        expect(imageElement).toBeInTheDocument();
        expect(imageElement).toHaveAttribute('src', props.image);
    });

    it('Renderizar com os props obrigatorios', () => {
        const { title, description, image, link } = props;

        const { container } = render(<Card title={title} description={description} image={image} link={link} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('Verificar se a classe CSS está definida', () => {
        render(<Card {...props} />);

        const cardElement = screen.getByTestId('card');
        expect(cardElement).toHaveClass('flex flex-col bg-[#ededed] md:bg-[#f5f5f5] shadow-md rounded-3xl py-12 px-9 xs400:px-12 transition duration-300 ease-in-out transform hover:bg-white mb-10 max-w-lg h-86 hover:shadow-[0_4px_4px_0px_rgba(0,0,0,0.3)]');

        const cardElementConteudo = screen.getByTestId('card_conteudo');
        expect(cardElementConteudo).toHaveClass('mb-4');

        const cardElementConteudoImagem = screen.getByTestId('card_conteudo_imagem');
        expect(cardElementConteudoImagem).toHaveClass('min-h-5 max-w-[300px]');

        const cardElementConteudoTitulo = screen.getByTestId('card_conteudo_titulo');
        expect(cardElementConteudoTitulo).toHaveClass('text-3xl font-bold mt-5 text-secondary');

        const cardElementConteudoDescricao = screen.getByTestId('card_conteudo_descricao');
        expect(cardElementConteudoDescricao).toHaveClass('text-secondary text-base mt-5 hyphens-auto text-justify');

        const cardBotao = screen.getByTestId('card_botao');
        expect(cardBotao).toHaveClass('flex justify-center items-center mt-auto');

        const cardBotaoLink = screen.getByTestId('card_botao_link');
        expect(cardBotaoLink).toHaveClass('flex gap-4 bg-secondary hover:bg-white hover:cursor-pointer hover:text-secondary hover:border hover:border-secondary text-white font-bold py-2 px-5 xs400:px-10 rounded-tr-3xl rounded-bl-2xl transition-all duration-300 ease-in-out');

    });

    it('Verifica se as funcionlidades funcionam', () => {
        render(<Card {...props} />);

        const cardBotaoLink = screen.getByTestId('card_botao_link');
        expect(cardBotaoLink).toHaveAttribute('href', props.link);

        //Testa se o botão está clicável e redirecionando para as paginas
        fireEvent.click(cardBotaoLink);
        expect(window.location.href).toBe(props.link);
    });
});
