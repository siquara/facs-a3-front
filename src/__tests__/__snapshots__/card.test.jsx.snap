// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Renderizar com os props obrigatorios 1`] = `
<div
  class="flex flex-col bg-[#ededed] md:bg-[#f5f5f5] shadow-md rounded-3xl py-12 px-9 xs400:px-12 transition duration-300 ease-in-out transform hover:bg-white mb-10 max-w-lg h-86 hover:shadow-[0_4px_4px_0px_rgba(0,0,0,0.3)]"
  data-testid="card"
>
  <div
    class="mb-4"
    data-testid="card_conteudo"
  >
    <div
      class="flex items-center gap-2"
    >
      <img
        class="min-h-5 max-w-[300px]"
        data-testid="card_conteudo_imagem"
        src="teste.png"
      />
    </div>
    <h1
      class="text-3xl font-bold mt-5 text-secondary"
      data-testid="card_conteudo_titulo"
    >
      Titulo Teste
    </h1>
    <p
      class="text-secondary text-base mt-5 hyphens-auto text-justify "
      data-testid="card_conteudo_descricao"
    >
      Descrição Teste
    </p>
  </div>
  <div
    class="flex justify-center items-center mt-auto"
    data-testid="card_botao"
  >
    <a
      class="flex gap-4 bg-secondary hover:bg-white hover:cursor-pointer hover:text-secondary hover:border hover:border-secondary text-white font-bold py-2 px-5 xs400:px-10 rounded-tr-3xl rounded-bl-2xl transition-all duration-300 ease-in-out"
      data-testid="card_botao_link"
      href="http://localhost/"
      target="_blank"
    >
      Acesse aqui
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M221.66,133.66l-72,72a8,8,0,0,1-11.32-11.32L196.69,136H40a8,8,0,0,1,0-16H196.69L138.34,61.66a8,8,0,0,1,11.32-11.32l72,72A8,8,0,0,1,221.66,133.66Z"
        />
      </svg>
    </a>
  </div>
</div>
`;
